plugin/evm/vm_warp_test.go

```diff
@@ -408,74 +410,142 @@ func TestReceiveWarpMessage(t *testing.T) {
 	}
 	genesisJSON, err := genesis.MarshalJSON()
 	require.NoError(err)
-	issuer, vm, _, _ := GenesisVM(t, true, string(genesisJSON), "", "")
+
+	// disable warp so we can re-enable it with RequirePrimaryNetworkSigners
+	disableTime := upgrade.InitiallyActiveTime.Add(10 * time.Second)
+	disableConfig := warp.NewDisableConfig(utils.TimeToNewUint64(disableTime))
+
+	// re-enable warp with RequirePrimaryNetworkSigners
+	reEnableTime := disableTime.Add(10 * time.Second)
+	reEnableConfig := warp.NewDefaultConfig(utils.TimeToNewUint64(reEnableTime)).WithRequirePrimaryNetworkSigners(true)
+
+	upgradeConfig := params.UpgradeConfig{
+		PrecompileUpgrades: []params.PrecompileUpgrade{
+			{Config: disableConfig},
+			{Config: reEnableConfig},
+		},
+	}
+	upgradeBytes, err := json.Marshal(upgradeConfig)
+	require.NoError(err)
+
+	issuer, vm, _, _ := GenesisVM(t, true, string(genesisJSON), "", string(upgradeBytes))
 
 	defer func() {
 		require.NoError(vm.Shutdown(context.Background()))
 	}()
 
-	acceptedLogsChan := make(chan []*types.Log, 10)
-	logsSub := vm.eth.APIBackend.SubscribeAcceptedLogsEvent(acceptedLogsChan)
-	defer logsSub.Unsubscribe()
+	// Subnet messages should use subnet signers
+	testReceiveWarpMessage(t, issuer, vm, false, false, upgrade.InitiallyActiveTime)
 
-	payloadData := avagoUtils.RandomBytes(100)
+	// At genesis, primary network messages should use subnet signers
+	blockTime := upgrade.InitiallyActiveTime.Add(2 * time.Second) // for fees
```

Comment:
`// for fees`

What does this mean?

Reply:
adding a comment to explain it.
for more context, avalanche dynamic fees assumes a 2s block time, while blocks can be issued faster (even at the same time), the fees will exceed what's specified in the UT transaction.